python/ray/tune/examples/tf_distributed_keras_example.py

```diff
@@ -49,16 +51,21 @@ def train_mnist(config, checkpoint_dir=None):
     multi_worker_dataset = mnist_dataset(global_batch_size)
     with strategy.scope():
         multi_worker_model = build_and_compile_cnn_model(config)
+    # Define the checkpoint directory to store the checkpoints
 
+    checkpoint_dir = "./training_checkpoints"
+    # Name of the checkpoint files
+    checkpoint_prefix = os.path.join(checkpoint_dir, "ckpt_{epoch}")
     multi_worker_model.fit(
         multi_worker_dataset,
         epochs=2,
         steps_per_epoch=70,
         callbacks=[
-            TuneReportCheckpointCallback(
-                {
-                    "mean_accuracy": "accuracy"
-                }, filename="checkpoint")
+            tf.keras.callbacks.ModelCheckpoint(
+                filepath=checkpoint_prefix, save_weights_only=True),
+            TuneReportCallback({
+                "mean_accuracy": "accuracy"
+            })
```

Comment:
@oliverhu can you explain what you're doing with this change? 

This won't trigger the Tune checkpointing mechanism (which requires a call to `tune.checkpoint_dir`).

Reply:
🤦  `TuneReportCheckpointCallback ` doesn't work under distributed training (not always on worker 0). Apparently switching to `ModelCheckpoint` callback out of box is not compatible with Tune. Let me update this.

Reply:
found the issue..seems some file name escaping problem

Reply:
actually not, distributed torch has the same issue.. @richardliaw did you see this before?

```
2020-11-15 07:09:52,540	ERROR trial_runner.py:712 -- Trial WrappedDistributedTorchTrainable_88060_00003: Error handling checkpoint /home/ray1/ray_results/WrappedDistributedTorchTrainable_2020-11-15_07-09-41/WrappedDistributedTorchTrainable_88060_00003_3_2020-11-15_07-09-41/checkpoint_10/./
Traceback (most recent call last):
  File "/home/ray1/.local/share/virtualenvs/ray1-OvGpSAQ1/lib/python3.6/site-packages/ray/tune/trial_runner.py", line 707, in _process_trial_save
    checkpoint=trial.saving_to)
  File "/home/ray1/.local/share/virtualenvs/ray1-OvGpSAQ1/lib/python3.6/site-packages/ray/tune/callback.py", line 204, in on_checkpoint
    callback.on_checkpoint(**info)
  File "/home/ray1/.local/share/virtualenvs/ray1-OvGpSAQ1/lib/python3.6/site-packages/ray/tune/syncer.py", line 450, in on_checkpoint
    self._sync_trial_checkpoint(trial, checkpoint)
  File "/home/ray1/.local/share/virtualenvs/ray1-OvGpSAQ1/lib/python3.6/site-packages/ray/tune/syncer.py", line 426, in _sync_trial_checkpoint
    trial, checkpoint.value))
ray.tune.error.TuneError: Trial WrappedDistributedTorchTrainable_88060_00003: Checkpoint path /home/ray1/ray_results/WrappedDistributedTorchTrainable_2020-11-15_07-09-41/WrappedDistributedTorchTrainable_88060_00003_3_2020-11-15_07-09-41/checkpoint_10/./ not found after successful sync down.
```

Reply:
it happens when driver is in remote host

Reply:
ok, it is 100% reproducible.. should be easy to fix

Reply:
Seems some regression in checkpoint logic.. even single node's checkpoint doesn't work.
```
2020-11-15 07:39:25,887	ERROR trial_runner.py:712 -- Trial TrainMNIST_9b13f_00019: Error handling checkpoint /home/ray1/ray_results/TrainMNIST_2020-11-15_07-38-51/TrainMNIST_9b13f_00019_19_lr=0.06339,momentum=0.2118_2020-11-15_07-39-19/checkpoint_16/model.pth
Traceback (most recent call last):
  File "/home/ray1/.local/share/virtualenvs/ray1-OvGpSAQ1/lib/python3.6/site-packages/ray/tune/trial_runner.py", line 707, in _process_trial_save
    checkpoint=trial.saving_to)
  File "/home/ray1/.local/share/virtualenvs/ray1-OvGpSAQ1/lib/python3.6/site-packages/ray/tune/callback.py", line 204, in on_checkpoint
    callback.on_checkpoint(**info)
  File "/home/ray1/.local/share/virtualenvs/ray1-OvGpSAQ1/lib/python3.6/site-packages/ray/tune/syncer.py", line 450, in on_checkpoint
    self._sync_trial_checkpoint(trial, checkpoint)
  File "/home/ray1/.local/share/virtualenvs/ray1-OvGpSAQ1/lib/python3.6/site-packages/ray/tune/syncer.py", line 426, in _sync_trial_checkpoint
    trial, checkpoint.value))
ray.tune.error.TuneError: Trial TrainMNIST_9b13f_00019: Checkpoint path /home/ray1/ray_results/TrainMNIST_2020-11-15_07-38-51/TrainMNIST_9b13f_00019_19_lr=0.06339,momentum=0.2118_2020-11-15_07-39-19/checkpoint_16/model.pth not found after successful sync down.
```

Reply:
Discussed with Richard offline, it is actually caused there is no SSH access between VMs. Checkpoints won't be synced if SSH channel is not built between the hosts.

Reply:
nice :)